include/kimera_vio_ros/RosDataProviderInterface.hpp

```diff
@@ -48,12 +65,14 @@ struct PointNormalUV {
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 } EIGEN_ALIGN16;
 
-class RosDataProviderInterface : public DataProviderInterface {
+class RosDataProviderInterface : public DataProviderInterface, public rclcpp::Node{
```

Comment:
:+1:

Reply:
Yea, I think composing these classes gives a more convenient public interface. I'd like to break out a bit more, such that the user facing ROS node class would inherit from two child classes, one for the input data acquisition (imu/camera callbacks), and one for the output publications. This might help improve code sharing between online and offline parcer interfaces, as well as for different modes of acquisition, stereo vs monocular setups.